### PR TITLE
Also allow jms/serializer-bundle 3.0 as both 2.0 and 3.0 are compatible with Sylius

### PR DESCRIPTION
**Pull request by reyostallenberg**
| Q               | A
| --------------- | -----
| Branch?         | 1.7
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/11301
| License         | MIT

### CODE REVIEW

**Comment by lchrusciel** [comment]
Sure, thanks for your effort! ;) 

As you can see in #10928, we will need to update them anyway. It may happen, that because of dependencies constraints, we will support both versions to enable Symfony 4.4 and Symfony 5.0 support.

File: composer.json [hunk 1]
```diff
@@ -97,8 +97,8 @@
         "webmozart/assert": "^1.1",
         "white-october/pagerfanta-bundle": "^1.0.8",
         "willdurand/hateoas": "^2.12",
-        "willdurand/hateoas-bundle": "^1.2",
-        "winzou/state-machine-bundle": "^0.3",
+        "willdurand/hateoas-bundle": "^1.2 || ^2.0",
+        "winzou/state-machine-bundle": "^0.3 || ^4.0",
```
Comment (pamil) [inline composer.json:101]: I guess this change can be reverted, there's no `winzou/state-machine-bundle` in `^4.0`.

Comment (pamil) [reply composer.json]: I removed the last commit from your PR, so we can get it merged faster 🎉

**Comment by pamil** [comment]
ResourceBundle needs to be updated first, right now the next major versions are not installed and tested.

**Comment by reyostallenberg** [comment]
I've made a mistake, I meant

https://github.com/winzou/StateMachineBundle/tree/0.4.2

What requirement is needed for ResourceBundle?

**Comment by pamil** [comment]
@reyostallenberg as for the next minor of WinzouStateMachineBundle, it would require some changes to provide backwards-compatibility - there are end-user Sylius applications that will break when we upgrade, the names of the services have changed in that release.

**Comment by reyostallenberg** [comment]
All right, that's not what we want. But for my current setup in an existing sf app I only got it working by requiring:
`"winzou/state-machine-bundle": "0.4.2 as 0.3.1"`

I need some time to sort out what we need to do with this dependency hell. Only allowing `"jms/serializer-bundle": "^2.0 || ^3.0"` is not enought to get all stuff working/installable.

**Comment by esserj** [comment]
@pamil what needs to happen here still? Because we ran into the same issue when trying to combine sylius with our bundles, I see the resource bundle does this:

```php
final class WinzouStateMachinePass implements CompilerPassInterface
{
    /**
     * {@inheritdoc}
     */
    public function process(ContainerBuilder $container): void
    {
        $container->setAlias('sm.factory', FactoryInterface::class);
        $container->setAlias('sm.callback_factory', CallbackFactoryInterface::class);
        $container->setAlias('sm.callback.cascade_transition', CascadeTransitionCallback::class);

        foreach (['sm.factory', 'sm.callback_factory', 'sm.callback.cascade_transition'] as $id) {
            try {
                $container->findDefinition($id)->setPublic(true);
            } catch (ServiceNotFoundException $exception) {
            }
        }
    }
}
```

Is this maybe something we can add to all bundles that support .3 and .4?

**Comment by esserj** [comment]
any feedback?

**Comment by reyostallenberg** [comment]
I think after all upgrades done this year, this issue can and should be closed

**Pull request closed without merging**